Automattic/wp-calypso
# Jetpack Cloud: Add app sidebar barebones

This PR adds barebones for the Jetpack Cloud sidebar.

It reuses the 'default' Calypso sidebar layout components such as `<Sidebar />`, `<SidebarMenu />` or `<ExpandableSidebarMenu />`. With this PR these components now accept the Material Icon's `style` property. This way we can use the `filled` style of the icons needed In Jetpack Cloud sidebar.

The currently expanded menus are stored in the component's state.

<img width="574" alt="Screenshot 2020-01-28 at 13 15 52" src="https://user-images.githubusercontent.com/478735/73263206-54353e80-41d0-11ea-992a-b7973ef96557.png">

#### Changes proposed in this Pull Request

* Add Jetpack Cloud sidebar barebones
* Move the sidebar file to the components directory
* Add support for different material icon styles in sidebar elements
* Add new SVG icons to material-design-icons package
* Rebuild SVG sprite in material-design-icons package
* Normalize routes in router so that we never end up e.g. with a trailing slash

#### Testing instructions



* Start Calypso with this command: `CALYPSO_ENV=jetpack-cloud-development npm start`
* Go to http://jetpack.cloud.localhost:3000/
* Try toggling both sidebar expandable menus
* Navigate to "Backups" -> "Backups"
* Confirm that the "Backups" menu item is highlighted
* Refresh the page and confirm the selection is still there and that the "Backups" menu is expanded by default
* Navigate to "Scan" -> "Scanner" and repeat the previous two steps to confirm the functionality works for other menus as well
* Navigate to "Dashboard" and confirm that the menu item is highlighted

**Please note that other menu items will not work at this stage since the paths and components are not yet defined in controller.**

Fixes n/a
